wagl/modtran.py

```diff
@@ -704,85 +718,54 @@ def read_spectral_response(fname, as_list=False, spectral_range=None):
-                             ('mlflx', 'int32'),
-                             ('iv1', 'float32'),
-                             ('band_width', 'float32'),
-                             ('fwhm', 'float32'),
-                             ('ifwhm', 'float32')])
-
-    # datatype for the dataframe containing the flux data
-    flux_dtype = numpy.dtype([('upward_diffuse', 'float64'),
-                              ('downward_diffuse', 'float64'),
-                              ('direct_solar', 'float64')])
-
-    with open(fname, 'rb') as src:
-        # read the hdr record
-        hdr_data = numpy.fromfile(src, hdr_dtype, count=1)
-
-        # maximum flux levels at a spectral grid point
-        levels = hdr_data['mlflx'][0] + 1
-
-        # define a datatype to read a record containing flux data
-        dtype = numpy.dtype([('wavelength', 'float64'),
-                             ('flux_data', 'float64', (levels, 3))])
-
-        # read the rest of the hdr which contains the altitude data
-        altitude = numpy.fromfile(src, dtype='float32', count=levels)
-
-        # read the record length end value
-        _ = numpy.fromfile(src, 'int32', count=1)
-
-        # initialise the FORTRAN read
-        ffile = FortranFile(src)
-
-        # read data from 2600 down to 350
-        flux = {}
-        wavelength_steps = range(2600, 349, -1)
-        for wv in wavelength_steps:
-            data = ffile.read_record(dtype)
-            df = pd.DataFrame(numpy.zeros(levels, dtype=flux_dtype))
-            #df['wavelength'] = data['wavelength'][0]
-            df['upward_diffuse'] = data['flux_data'].squeeze()[:, 0]
-            df['downward_diffuse'] = data['flux_data'].squeeze()[:, 1]
-            df['direct_solar'] = data['flux_data'].squeeze()[:, 2]
-            flux[wv] = df
-
-    # concatenate into a single table
-    flux_data = pd.concat(flux, names=['wavelength', 'level'])
-
-    # setup a dataframe for the altitude
-    altitude = pd.DataFrame({'altitude': altitude})
-    altitude.index.name = 'layer'
-
-    return flux_data, altitude
-
-
-def read_modtran_channel(fname, acquisition, albedo):
+    
+    f = open(tp6_fname,'r')
+    cnt = 0
+    cnt_start = 0
+
+    for line in f.readlines(): 
+        cnt +=1 
+        if fnmatch.fnmatch(line,'*SINGLE SCATTER SOLAR PATH GEOMETRY TABLE FOR MULTIPLE SCATTERING*'):
+            cnt_start = cnt
+            continue
+
+    f.close()
+    f = open(tp6_fname,'r')
+    
+    lines = list(f.readlines()[cnt_start+4: cnt_start + 40])  # this constant is (40-4 = num of atm 
+                                                              # layer)
+    data = numpy.vstack([ln.rstrip().split() for ln in lines]).T
+      
+    solar_zenith = data[3] # solar zenith angle at all atmosphere layers
+    
+    return solar_zenith
+
+    
+def read_modtran_channel(fname, tp6_fname, acquisition, albedo):
     """
     Read a MODTRAN output `*.chn` ascii file.
-
+    
```

Comment:
Not sure what has happened here, but it looks like a TAB has been included.  Can you check your IDE settings to not use TAB's.

Reply:
check